### DISCUSSION TITLE
List support

### DISCUSSION THREAD

**Discussion by akosholloszabo** [AndreiKingsley/lib-ggdsl#377] [Ideas]
I love the library. It is already way better than plotly. However every time I load a data that is not stored in a dataframe or csv it is rearly painful to convert from a per row data class based structure to a column based one. It would be very useful if I could call it for any list and in each lambda of the DSL I could say how to map the record to the to the value relevant for the lambda. 

For example if I have the data class:
`
data class C(
 val x: String,
val y: Int,
val z:String
)
`

My data would be a List<C> and I would use it like.
`groupby{it.z}` or
`x{it.x}` or
`y{it.y}` or
`color{it.z}` 

I hope I could explain my idea well. Let's talk about it!

**Comment by AndreiKingsley** [comment]
Hi! Thank you ❤️ .
Yes, we have considered adding a row api (similar to what we already have in DataFrame - see [row expressions](https://kotlin.github.io/dataframe/datarow.html#row-expressions)), but there are a few interface and implementation issues. In particular, the lets-plot uses a system of markers - that is, pointers to the columns of the dataframe (using column names), and these pointers are used in many parts of the api; this would be more difficult to implement via row elements referencing. However, we are still thinking about the idea - moreover, a simple version of such api is quite realistic to implement..

**Comment by AndreiKingsley** [comment]
Now, the best solution for your case, as I see:
1) Cast your list to DataFrame using `.toDataFrame()`
2) Use `KProp` API in DSL (it's not supported everywhere in DSL yet, but that's easily fixable)
<img width="877" alt="image" src="https://github.com/Kotlin/kandy/assets/49092338/82e01a86-6058-4b9a-b6d4-24f110b30bb6">
